### DISCUSSION TITLE
Type hints for Bidict type

### DISCUSSION THREAD

**Discussion by BD103** [jab/bidict#221] [Q&A] [answered]
Hello,

I am creating a Python package that uses bidict. I'm wondering what type hinting I should assign to this data.

```python
b: bidict[str, str] = bidict(
  {
    "H": "hydrogen",
    "C": "carbon"
  }
)
```

Is the `bidict[str, str]` accurate? Thank you.

**Comment by jab** [comment, marked as answer]
Yes, that is correct. You can use `reveal_type` to have mypy confirm this for you:
```python
# demo.py
from bidict import bidict

b = bidict({"H": "hydrogen", "C": "carbon"})
reveal_type(b)
```
```sh
$ mypy demo.py
demo.py:4: note: Revealed type is "bidict._bidict.bidict[builtins.str*, builtins.str*]"
```

What package are you creating that uses bidict?

**Comment by BD103** [comment]
My personal Python package uses bidict for hexadecimal conversion. You can see the work-in-progress code [here](https://github.com/BD103/BD103-Python/blob/87dd0d9bdb2bb0fa16c75d5adb332e1700dcc57f/src/bd103/data/hex.py) if you're interested. Either way, thanks for the confirmation! :D
